widgets: add onclick feature

PaideiaDilemma opened the pull request:
Picks up the contribution from @Memoraike (thanks :))
Implements https://github.com/hyprwm/hyprlock/issues/366.

PaideiaDilemma commented:
Currently `on-click` is implemented for labels and onHover changes the cursor when on the input-field.

Do we want to implement the following things here already or in a future MR?
- `on-click` for shape and image.
- visual feedback for onClick (and maybe for onHover)

rotmh commented:
I don't have any say here... :)

But in my opinion:

> `on-click` for shape and image.

Should probably be implemented in this MR, because it's part of the same premise as `on-click` for labels, but:

> visual feedback for onClick (and maybe for onHover)

I think should be a part of a separate MR, because it's seems like a separate feature.

PaideiaDilemma commented:
Added onclick for shape and image.

Then i noticed the it triggers onclick for widgets on different monitors than my cursor is on. (I have two 1080p monitors and a single shape widget would trigger the onclick command twice)
I implemented handling for the `enter` and `leave` events of `wl_pointer` to localize onclick and hover to the currently focused output and fixed up the positions a bit.

PaideiaDilemma commented:
ups, sorry. didn't mean to delete the branch

JunaidQrysh commented:
@PaideiaDilemma can you please add visual feedback when cursor is on a label or image; i have setup reboot icon
it would be good if it says "Reboot" when the cursor is on that icon.

vaxerski commented on src/core/CursorShape.cpp:
```diff
@@ -9,8 +9,13 @@ CCursorShape::CCursorShape(SP<CCWpCursorShapeManagerV1> mgr) : mgr(mgr) {
 
 void CCursorShape::setShape(const wpCursorShapeDeviceV1Shape shape) {
-    if (!dev)
+    if (!g_pSeatManager->m_pPointer) {
```
no {} here and blow

PaideiaDilemma commented:
@JunaidQrysh  yeah i want that as well but not in this PR. We can add it later

vaxerski commented on src/renderer/widgets/Shape.cpp:
```diff
@@ -100,3 +103,19 @@ bool CShape::draw(const SRenderData& data) {
+
+void CShape::onClick(uint32_t button, bool down, const Vector2D& pos) {
+    if (down && !onclickCommand.empty())
+        g_pHyprlock->spawnSync(onclickCommand);
```
dumb idea. someone puts `htop` by accident and hyprlock is stuck

evtn replied:
the current [dynamic labels](https://wiki.hyprland.org/Hypr-Ecosystem/hyprlock/#dynamic-labels) doc states that one shouldn't put never-ending commands there already, how is this different?

PaideiaDilemma replied:
dynamic labels run in the asyncResourceGatherer.
Making that thread hang is also not ideal but not as bad as stalling the "main" thread (which handles all wayland events, etc.)

vaxerski replied:
furthermore there is no need for sync here so it's just hanging for no reason

vaxerski reviewed (approved):
at your discretion
